File: server/scopelens-server/routers/team.go
package routers

import (
	"github.com/gin-gonic/gin"
	v1 "scopelens-server/api/v1"
	"scopelens-server/middleware"
)

func InitTeamAUTHRouter(Router *gin.RouterGroup) {
	TeamRouter := Router.Group("team").Use(middleware.JWTAuth())
	{
		TeamRouter.POST("/post", v1.InsertTeam)
	}
}

func InitTeamRouter(Router *gin.RouterGroup) {
	TeamRouter := Router.Group("team")
	{
		TeamRouter.GET("/teams", v1.GetTeams)
		TeamRouter.GET("/likes", v1.GetTeamsOrderbyLikes)
		TeamRouter.GET("/teams/:id", v1.GetTeamByID)
		TeamRouter.POST("/search", v1.GetTeamsBySearchCriteria)
	}
}

File: web/scopelens-website/src/api/team.js
import http from "./index";

export const insertTeam = (data, token) => {
    const headers = {
        'Content-Type': 'application/json',
        'Authorization': token
    }
    return http.post("team/post", data, {
        headers: headers
    });
}

export const getTeams = (page) => {
    return http.get("/team/teams", {
        params: {
            page: page
        }
    })
};

export const getTeamByID = (id) => {
    return http.get("/team/teams/" + id)
};

export const getTeamsBySearchCriteria = (page, data) => {
    const headers = {
        'Content-Type': 'application/json',
    }
    return http.post("team/search", data, {
        headers: headers,
        params: {
            page: page
        }
    });
};